### ISSUE TITLE
[V3] posibility to increase Provider state callback timeout

### ISSUE THREAD

**Issue by individual-it** [pact-foundation/pact-js#600]
### Checklist
Before making a feature request, I have: 

- [x] [Searched the issues to check that this feature hasn't been requested before](https://github.com/pact-foundation/pact-js/issues?q=is%3Aissue)
- [x] Checked the documentation to see if it is possible to do what I want already 

### Feature description

The timeout for setting states seems to be fixed to 1000ms it would be great to be able to increase that timeout

### Use case

depending on the provider, the state that need to be set and the hardware it might take longer than 1sec to set the state e.g when cleaning up or resetting a provider

**Comment by individual-it** [comment]
If I'm not wrong that was implemented by @rholshausen for pact-reference in https://github.com/pact-foundation/pact-reference/commit/4afa86a785686d3bd5aa16c3c1fff17969613948

**Comment by individual-it** [comment]
looks like there is work done in this issue before I've even created it e.g. https://github.com/pact-foundation/pact-js/commit/727968ca85dc33bc1569e6a764fe049d4f01645a :+1: 

so just waiting for the next release :zzz:

**Comment by uglyog** [comment]
Still need someone to do this bit:
```
            // TODO: add this in once the downstream lib has been released
            // timeout: self.options.callback_timeout
```
Hopefully @mefellows does it :-D

**Comment by individual-it** [comment]
@uglyog the current master of pact-js already helps because it has 10sec timeout hardcoded ;-)

**Comment by mefellows** [comment]
On it. I'll have this flag in the next release (tonight, assuming I don't mess up the release step).

**Comment by individual-it** [comment]
fixed by e891fccb5802491cf148f398c81406bf06ae43c8
